### PR TITLE
Added logging args for level and file

### PR DESCRIPTION
**Pull request by sschleemilch**
#  About

Improved logging capabilities with:
- Added argument for controlling the log levels via `--log-level` with "INFO" as default
- Added argument for writing logs additionally into a log file with `--log-file`
- Using a `log` instance

Unrelated:
- Cleaned up docs to be up to date with changes regarding poetry
- Added code syntax highlighting for all code blocks in docs
- Using [alerts](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts)

### CODE REVIEW

File: src/vss_tools/vspec/vspec2x.py [hunk 1]
```diff
@@ -43,9 +44,11 @@ def __init__(self, generator: Vss2X, vspec2vss_config: Vspec2VssConfig):
         self.vspec2vss_config = vspec2vss_config
 
     def main(self, arguments):
-        initLogging()
         parser = argparse.ArgumentParser(description="Convert vspec to other formats.")
 
+        parser.add_argument("--log-level", choices=["INFO", "DEBUG", "WARNING", "ERROR", "CRITICAL"],
```
Comment (erikbosch) [inline src/vss_tools/vspec/vspec2x.py:49]: Could you take a look at https://github.com/COVESA/vss-tools/blob/master/docs/vspec2x.md and add some info there as well on the new arguments.

(And yes - I know that page is a bit messy so if feel free to more than just adding the two new args there if you like :-) )

Comment (sschleemilch) [reply src/vss_tools/vspec/vspec2x.py:49]: Sure, will do

Comment (sschleemilch) [reply src/vss_tools/vspec/vspec2x.py:49]: Done. Tried to clean up and modernize all docs in the process.

**Review by erikbosch — APPROVED** [review]
Looks good to me. 

Ask on next VSS meeting if anyone more is interested in review it, if not I do not think keeping it open one more week is needed

**Pull request merged**